include/utils/utils.hh

```diff
@@ -0,0 +1,175 @@
+#ifndef __INCLUDE_UTILS_UTILS_HH__
+#define __INCLUDE_UTILS_UTILS_HH__
+
+#include <cassert>
+#include <concepts>
+#include <iterator>
+
+namespace utils
+{
+template <class T>
+concept ItContainer = requires(T cont)
+{
+  {
+    std::begin(cont)
+    } -> std::forward_iterator;
+  {
+    std::end(cont)
+    } -> std::forward_iterator;
+  {
+    std::size(cont)
+    } -> std::convertible_to<std::size_t>;
+  {
+    typename std::remove_reference_t<T>::iterator()
+    } -> std::forward_iterator;
```

Comment:
Won't work if iterator class named as `Iterator` for example

Reply:
Fixed